docs/proposals/ha-configuration.md

```diff
@@ -0,0 +1,186 @@
+# High Availability (HA) Configuration
+
+## Problem
+In the current OpenShift operational flow, one or more HAProxy routers are
+used to direct network traffic to the target services. This makes use of
+HAProxy as a layer-7 load balancer to reach multiple services/applications.
+This require the HAProxy routers to be "discoverable" - meaning the IP
+address resolution scheme needs to know where each HAProxy router is
+running. This "discovery" mechanism in most cases would be via DNS but
+that doesn't preclude using some other service discovery mechanism ala
+via zookeeper or etcd.
+
+In any case, that methodology works fine in steady state conditions.
+It does, however have implications on failure conditions where the machine
+running the router process goes down or the router process dies or there
+is a network split. When these failure conditions arise, the above model
+breaks down as it requires the caller (or something in the caller's
+execution chain) to implement either a health checker or retry failed
+requests and/or update the "discovery" mechanism to remove the failed
+instance(s) from the traffic mix. Otherwise, a certain subset of requests
+(failed/total) will fail via the upstream "discovery" mechanism (e.g. DNS).
+
+
+## Use Cases
+  1. As an administrator, I want my cluster to be assigned a resource set
+     and I want the cluster to automatically manage those resources.
+  2. As an administrator, I want my cluster to be assigned a set of virtual
+     IP addresses that the cluster manages and migrates (with zero or
+     minimal downtime) on failure conditions.
+  3. As an addendum to use case #2, the administrator should not be
+     required to perform any manual interaction to update the upstream
+     "discovery" sources (e.g. DNS). The cluster should service all the
+     assigned virtual IPs when atleast a single node is available - and
+     this should be inspite of the fact that the current available
+     resources are not sufficient to reach "critical mass" aka the
+     desired state.
+
+
+## Goals
+The goal here is to provide the OpenShift environment with one or more
```

Comment:
Solution is specific to node ips.  What would it take to give pods stable VIPs?  Can the ipfailover pod handle VIPs for sets of pod ips?  What changes would be required to support both and how much work would it be?

How does Neil's external ip solution affect this setup?  If we have routers an external ip, would this be able to manage that?